pkg/operator/baremetal_pod.go

```diff
@@ -37,29 +37,11 @@ var volumeMounts = []corev1.VolumeMount{
 	},
 }
 
-func buildEnvVarFromConfigMap(name string, key string) corev1.EnvVar {
-	return corev1.EnvVar{
-		Name: name,
-		ValueFrom: &corev1.EnvVarSource{
-			ConfigMapKeyRef: &corev1.ConfigMapKeySelector{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: baremetalConfigmap,
-				},
-				Key: key,
-			},
-		},
-	}
-}
-
 func buildEnvVar(name string, key string, baremetalProvisioningConfig BaremetalProvisioningConfig) corev1.EnvVar {
 	value := getMetal3DeploymentConfig(name, baremetalProvisioningConfig)
-	if value != nil {
-		return corev1.EnvVar{
-			Name:  name,
-			Value: *value,
-		}
-	} else {
-		return buildEnvVarFromConfigMap(name, key)
+	return corev1.EnvVar{
+		Name:  name,
+		Value: *value,
```

Comment:
wouldn't this panic if value is nil?

Reply:
getBaremetalProvisioningConfig() will return BaremetalProvisioningConfig{} if it is unable to read any config value from the Provisioning CR. If we end up with BaremetalProvisioningConfig{}, metal3 deployment will not proceed. So, when get to this point, we will not have value of nil.

Reply:
Fixed.